apps/crn-frontend/src/shared-research/export.ts

```diff
@@ -91,31 +94,33 @@ export const researchOutputToCSV = (
   lastModifiedDate: output.lastModifiedDate,
 });
 
-export const createCsvFileStream = (
-  csvOptions: Parameters<typeof format>[0],
-  fileName: string,
-) => {
-  const csvStream = format({ writeBOM: true, ...csvOptions });
+export const createCsvFileStream = (fileName: string, csvOptions?: Options) => {
   // If the WritableStream is not available (Firefox, Safari), take it from the ponyfill
   if (!window.WritableStream) {
     streamSaver.WritableStream = WritableStream;
   }
   const fileWriter = streamSaver.createWriteStream(fileName).getWriter();
-  csvStream
-    .on('data', (data) => fileWriter.write(data))
+  const stringifier = stringify({ bom: true, ...csvOptions });
+  return stringifier
+    .on('readable', () => {
+      let row;
+      while ((row = stringifier.read()) !== null) {
+        fileWriter.write(row);
+      }
+    })
     .on('end', () => fileWriter.close());
-  return csvStream;
 };
 
 export const algoliaResultsToStream = async <T extends keyof EntityResponses>(
-  csvStream: CsvFormatterStream<Row, Row>,
+  csvStream: Stringifier,
   getResults: ({
     currentPage,
     pageSize,
   }: Pick<GetListOptions, 'currentPage' | 'pageSize'>) => Readonly<
     Promise<SearchEntityResponse<T>>
   >,
-  transform: (result: EntityResponses[T]) => Row,
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  transform: (result: EntityResponses[T]) => Record<string, any>,
```

Comment:
This is a hack because of the algolia library work. I have just maintained the type from Fast-CSV (we now use CSV because they include relavant polyfills.)